src/App.tsx

```diff
@@ -1,8 +1,43 @@
+import { useState } from "react";
+
+import { DisplayCard } from "./components/DisplayCard/DisplayCard";
+
 const App = () => {
+	const [nameInputValue, setNameInputValue] = useState<string>("");
+	const [entities, setEntities] = useState<any[]>([]);
+
+	const queryString = nameInputValue.toLowerCase();
+	const fetchURL = `https://pokeapi.co/api/v2/pokemon/${queryString}/`;
+
+	const fetchEntity = async () => {
+		const response = await fetch(fetchURL);
+		const data = await response.json();
+		setEntities([...entities, data]);
+		setNameInputValue("");
+	};
+
 	return (
 		<>
 			<div>
-				<h1>Pokémon Battle</h1>
+				<input
+					value={nameInputValue}
+					type="text"
+					placeholder="Type a name..."
+					onKeyDown={(e) => {
+						if (e.key === "Enter") {
```

Comment:
If you wrap this input into a form element, and use the forms `onSubmit={fetchEntity}`   pressing enter in a field within a form automatically submits the form, so you don't need to rebuild that functionality like this.

Reply:
Ah, perfect. Ta.